src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.Build.ps1

```
@@ -1513,4 +1513,125 @@ class Build: ADOSVTBase
+                        {
+                            $branches | where-object {
+                                $currentBranch = $_    
+                                $refobj = "" | Select-Object branch,fileName
+                                $refobj.branch = $currentBranch
+                                try{
+                                    $url = 'https://dev.azure.com/{0}/{1}/_apps/hub/ms.vss-build-web.ci-designer-hub?pipelineId={2}&branch={3}&__rt=fps&__ver=2' -f $orgName, $projectId , $buildId, $currentBranch;
+                                    $responseObj = @([WebRequestHelper]::InvokeGetWebRequest($url);)
+                                    if([Helpers]::CheckMember($responseObj,"fps.dataProviders.data") -and $responseObj.fps.dataProviders.data.'ms.vss-build-web.pipeline-editor-data-provider' -and [Helpers]::CheckMember($responseObj.fps.dataProviders.data.'ms.vss-build-web.pipeline-editor-data-provider',"content") -and  $responseObj.fps.dataProviders.data.'ms.vss-build-web.pipeline-editor-data-provider'.content)
+                                    {
+                                        $dataprovider = $responseObj.fps.dataProviders.data.'ms.vss-build-web.pipeline-editor-data-provider'
+                                        $yamlFileContent = $dataprovider.content
+                                        if($yamlFileContent -match $regex)
+                                        {
+                                            $refobj.fileName = $dataprovider.definition.process.yamlFilename
+                                            $resultObj += $refobj
+                                        }
+                                    }
+                                }
+                                catch
+                                {
+                                    $controlResult.AddMessage([VerificationResult]::Error,"Not able to fetch YAML file for the branch: $($currentBranch)");
```

You should eat exception for this catch block, because even if there is any error then also control will pass in next else statement.

---

added another check to resolve this issue